### PR TITLE
Vulkan + Wayland + FreeBSD

### PR DESCRIPTION
**Pull request by valpackett**
Hi, thanks for finally adding desktop Vulkan support, this is awesome!!

I managed to get it running on my setup (FreeBSD 12-CURRENT + Weston compositor).

ppsspp-vulkan-working

This was mostly trivial, expect I've spent like a couple hours debugging a segfault that happened when trying to run `vkEnumeratePhysicalDevices`. That pointer just referenced some junk instead of actual driver code. Turns out it was `VULKAN_STATIC`'s fault, so I deleted it.

I suppose CMake's `FindVulkan` actually failed, then the `find_library` here successfully found the global Vulkan SDK instead of local, and set `-DVULKAN_STATIC`. (The binary was linked to `/usr/local/lib/libvulkan.so` and it was `dlopen`ing it too.)

You might want to ignore that last commit if you have a better solution in mind, but I don't know why you would ever link to `libvulkan`. Dolphin only `dlopens` it, and that works great. Especially for OS package maintainers who get one less dependency/option flag :)

Also, `PPSSPPSDL` should be made HiDPI aware on Wayland, I might look into that.

On Qt, it is DPI aware, but the image is flipped with Vulkan o_0

ppsspp-vulkan-qt-wayland-egl

(and the weird transparency always happens on Qt's `wayland-egl` QPA, even with OpenGL)

### CODE REVIEW

**Comment by myfreeweb** [comment]
oops, looks like I need to check for Wayland headers, I'll fix that

File: Common/Vulkan/VulkanLoader.cpp [hunk 1]
```diff
@@ -167,9 +165,13 @@ PFN_vkCmdExecuteCommands vkCmdExecuteCommands;
 PFN_vkCreateAndroidSurfaceKHR vkCreateAndroidSurfaceKHR;
 #elif defined(_WIN32)
 PFN_vkCreateWin32SurfaceKHR vkCreateWin32SurfaceKHR;
-#elif defined(VK_USE_PLATFORM_XLIB_KHR)
+#else
```
Comment (unknownbrackets) [inline Common/Vulkan/VulkanLoader.cpp]: This `#else` got paired with the wrong `#endif`, that's why Android is complaining.

-[Unknown]

**Review by unknownbrackets — CHANGES_REQUESTED** [review]
Cool.

I think that our Qt port doesn't actually support Vulkan yet.  That flipped thing is what would happen on Android when you selected Vulkan before we added support.

-[Unknown]

**Comment by hrydgard** [comment]
The local vulkan.so thing is from Sascha's Vulkan examples, https://github.com/SaschaWillems/Vulkan/ . I installed nvidia drivers on my laptop but dlopen-ing vulkan.so failed because the file doesn't seem to exist, but using the static vulkan library it works. However I guess I should fix my installation instead... any idea what apt package I need to install?

Regarding the flipped UI, something thinks it's OpenGL and uses an upside down projection matrix. Should be easy to fix.

**Comment by myfreeweb** [comment]
Seems like `libvulkan1`, `libvulkan-dev` on ubuntu

**Comment by mrcmunir** [comment]
Or install manually from lunarG https://vulkan.lunarg.com/sdk/home#linux

Download the .run and chmod a+x ./vulkansdk-linux-x86_64-1.0.65.0.run

**Review by unknownbrackets — APPROVED** [review]

**Comment by hrydgard** [comment]
Yup, this seems like the way to go. Merging.

**Pull request merged**